File: parsers/rust/examples/verify.rs
use kappa_parser::parse;
use std::fs;

fn main() {
    let dir = "../../examples/dense";
    let mut ok = 0;
    let mut total = 0;
    let mut entries: Vec<_> = fs::read_dir(dir).unwrap()
        .filter_map(|e| e.ok())
        .filter(|e| e.path().extension().map_or(false, |ext| ext == "kappa"))
        .collect();
    entries.sort_by_key(|e| e.file_name());
    
    for entry in &entries {
        total += 1;
        let src = fs::read_to_string(entry.path()).unwrap();
        let result = parse(&src);
        let fields: usize = result.entities.iter().map(|e| e.fields.len()).sum();
        if result.diagnostics.is_empty() {
            println!("PASS {}: {} entities, {} fields",
                entry.file_name().to_string_lossy(),
                result.entities.len(), fields);
            ok += 1;
        } else {
            println!("FAIL {}: {}", entry.file_name().to_string_lossy(),
                result.diagnostics[0].message);
        }
    }
    println!("{}/{} passed", ok, total);
    if ok != total { std::process::exit(1); }
}

File: parsers/java/src/main/java/dev/kappa/Verify.java
package dev.kappa;

import java.io.*;
import java.nio.file.*;
import java.util.*;

public class Verify {
    public static void main(String[] args) throws Exception {
        Path dir = Path.of("/tmp/kappa/examples/dense");
        int ok = 0, total = 0;
        List<Path> files = new ArrayList<>();
        try (var s = Files.list(dir)) {
            s.filter(p -> p.toString().endsWith(".kappa")).sorted().forEach(files::add);
        }
        for (Path p : files) {
            total++;
            String src = Files.readString(p);
            var result = KappaParser.parse(src);
            int fields = result.entities().stream().mapToInt(e -> e.fields().size()).sum();
            if (result.diagnostics().isEmpty()) {
                System.out.printf("PASS %s: %d entities, %d fields%n", p.getFileName(), result.entities().size(), fields);
                ok++;
            } else {
                System.out.printf("FAIL %s: %s%n", p.getFileName(), result.diagnostics().get(0).message());
            }
        }
        System.out.printf("%d/%d passed%n", ok, total);
        if (ok != total) System.exit(1);
    }
}
